src/@types/index.ts

```diff
@@ -14,8 +14,10 @@ type Card = {
 type CardCompany = (typeof CARD_COMPANY)[number] | '';
 
 type SetState = {
-  (value: string): void;
-  (key: string, value: string): void;
+  <T extends string>(value: T): void;
+  <T extends string>(value: T, key?: string): void;
```

Comment:
이렇게 되면 굳이 오버로딩을 하지 않아도 되지 않을까요?

```suggestion
  <T extends string>(value: T, key?: string): void;
```